### PR TITLE
[PVR] Misc performance improvements for PVR windows

### PR DESCRIPTION
**Pull request by ksooo**
@xhaggi mind taking a look

### CODE REVIEW

File: xbmc/addons/binary/interfaces/api1/PVR/AddonCallbacksPVR.cpp [hunk 1]
```diff
@@ -437,14 +437,11 @@ void CAddonCallbacksPVR::PVREpgEventStateChange(void* addonData, EPG_TAG* tag, u
...
-  CLog::Log(LOGDEBUG, "PVR - %s - state for epg event '%d' on channel '%d' on client '%s' changed to '%d'.",
-            __FUNCTION__, tag->iUniqueBroadcastId, iUniqueChannelId, client->Name().c_str(), newState);
-
   static CCriticalSection queueMutex;
   static std::vector<EpgEventStateChange> queuedChanges;
 
-  // during Kodi startup, addons may push updates very early, even before EPGs are ready to use.
-  if (g_PVRManager.EpgsCreated())
+  // during Kodi startup, addons may push updates very early, even before PVR manager and EPGs are ready to use.
+  if (g_PVRManager.IsStarted())
```
Comment (xhaggi) [inline xbmc/addons/binary/interfaces/api1/PVR/AddonCallbacksPVR.cpp]: not sure we should check both as the EPG gets loaded async and the PVR could be ready before the EPG was created.

Comment (ksooo) [inline xbmc/addons/binary/interfaces/api1/PVR/AddonCallbacksPVR.cpp]: yeah. actually I need to remove this commit as it is always the case that asynchronous epg creation (a CJob) gets triggered by pvr manager synchronously **after** pvr mananger sets itself to started. => https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/PVRManager.cpp#L537

**Comment by ksooo** [comment]
@xhaggi all your feedback is now incorporated. thx.

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp [hunk 2]
```diff
@@ -85,6 +85,13 @@ std::string CGUIWindowPVRBase::GetSelectedItemPath(bool bRadio)
   return m_selectedItemPaths[bRadio];
 }
 
+void CGUIWindowPVRBase::ResetObservers(void)
```
Comment (xhaggi) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: looks like an orphan method or do you call it anywhere else?
btw my comments are lost after your force-push.

File: xbmc/pvr/windows/GUIWindowPVRBase.h [hunk 3]
```diff
@@ -68,11 +68,14 @@ namespace PVR
...
     virtual bool OnBack(int actionID) override;
     virtual bool OpenGroupSelectionDialog(void);
-    virtual void ResetObservers(void) {};
     virtual void Notify(const Observable &obs, const ObservableMessage msg) override;
     virtual void SetInvalid() override;
     virtual bool CanBeActivated() const override;
 
+    void ResetObservers(void);
+    virtual void RegisterObservers(void) {};
+    virtual void UnregisterObservers(void) {};
```
Comment (xhaggi) [inline xbmc/pvr/windows/GUIWindowPVRBase.h]: any reason to make them public?

**Comment by xhaggi** [comment]
two of them are lost in github's force-push universe ;)

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp (hunk 2 above)
Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: not orphaned. =>  https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/PVRManager.cpp#L662

**Comment by xhaggi** [comment]
Great thank you :+1:

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp (hunk 2 above)
Comment (xhaggi) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: sorry for the noice, used in PVRManager ;)

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp [hunk 4]
```diff
@@ -181,11 +188,15 @@ void CGUIWindowPVRBase::OnInitWindow(void)
...
   m_viewControl.SetSelectedItem(GetSelectedItemPath(m_bRadio));
+
+  RegisterObservers();
 }
 
 void CGUIWindowPVRBase::OnDeinitWindow(int nextWindowID)
 {
+  UnregisterObservers();
   UpdateSelectedItemPath();
+  CGUIWindowPVRBase::OnDeinitWindow(nextWindowID);
```
Comment (FernetMenta) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: this crashes badly on my system. You call the same method again and again.

Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: @FernetMenta Ofc. Sorry, already fixed. You were just to fast. On my way to work while exchanging emails with you I realized that base class never was called here and I "fixed" this on the fly - with the result you noticed... One never should commit stuff without runtime testing. Shame on me. ;-)

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**